### PR TITLE
Update | Add Usul Proposals to Dashboard

### LINKED ISSUE #662: Merge Proposals history with Treasury Transactions in DAO homepage
### Overview

This list should show Treasury transactions and proposals history

### Link to documentation

https://www.figma.com/file/uRV3Nfowdzt5CknW1FgUkQ/%F0%9F%93%9A-Fractal-Design-System-%7C-Base-Components-%7C-October-2022?node-id=2158%3A12667&t=ztkrKXthwIf9Hvc0-0

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
Update DAO Dashboard with Usul Proposals

## Issue / Notion doc (if applicable)
Closes #662

## Screenshots (if applicable)
localhost_3000_ (5)

### CODE REVIEW

File: src/components/ui/badges/Badge.tsx [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
 
 type BadgeType = { [key: string]: { Icon: any; bg: string; color: string } };
 
-export enum BadgeLabels {
-  STATE_PENDING = 'statePending',
-  STATE_ACTIVE = 'stateActive',
-  STATE_PASSED = 'statePassed',
-  STATE_EXECUTED = 'stateExecuted',
-  STATE_REJECTED = 'stateRejected',
-  PARENT = 'parent',
```
Comment (mobiusTripper-crypto) [inline src/components/ui/badges/Badge.tsx:13]: i couldn't find PARENT or CHILD translations used anywhere so maybe they should be removed from the translation file.

File: src/pages/DaoDashboard/Activities/hooks/useActivities.ts [hunk 2]
```diff
@@ -120,18 +125,31 @@ export const useActivities = (sortBy: SortBy) => {
...
   const filterActivities = useMemo(() => {
     if (type === GovernanceTypes.GNOSIS_SAFE_USUL) {
-      return [...parsedActivities].filter(
-        activity => activity.eventType === ActivityEventType.Treasury
-      );
+      const txActivities = [...txProposals].map(proposal => {
+        const usulProposal = proposal as UsulProposal;
+
+        const activity: GovernanceActivity = {
+          eventDate: format(new Date(usulProposal.blockTimestamp * 1000), 'MMM dd yyyy'),
```
Comment (herbig) [inline src/pages/DaoDashboard/Activities/hooks/useActivities.ts:132]: we have `DEFAULT_DATE_FORMAT` constant that we could reuse here

File: src/components/ui/badges/Badge.tsx (hunk 1 above)
Comment (Da-Colon) [reply src/components/ui/badges/Badge.tsx:13]: I'll double check but these should be removed.

File: src/pages/DaoDashboard/Activities/hooks/useActivities.ts [hunk 3]
```diff
@@ -120,18 +125,31 @@ export const useActivities = (sortBy: SortBy) => {
    */
   const filterActivities = useMemo(() => {
     if (type === GovernanceTypes.GNOSIS_SAFE_USUL) {
-      return [...parsedActivities].filter(
-        activity => activity.eventType === ActivityEventType.Treasury
-      );
+      const txActivities = [...txProposals].map(proposal => {
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/pages/DaoDashboard/Activities/hooks/useActivities.ts:128]: when using array from other state variables, I've gotten in the practice of spreading the array as other wise I've seen weird side of effects of state not updating correctly

File: src/pages/DaoDashboard/Activities/hooks/useActivities.ts [hunk 4]
```diff
@@ -4,17 +4,21 @@ import {
...
+import { UsulProposal, TxProposalState } from './../../../../providers/Fractal/governance/types';
+import { Activity, GovernanceActivity } from './../../../../types/activity';
 
 export const useActivities = (sortBy: SortBy) => {
   const {
     gnosis: { safe, transactions },
-    governance: { type },
+    governance: {
+      type,
+      txProposalsInfo: { txProposals },
```
[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/pages/DaoDashboard/Activities/hooks/useActivities.ts:20]: yessir. It also contains data for pending and passed proposals

**Review by herbig — APPROVED** [review]
🚀

**Pull request merged**